File: assets/code-variants/ch12/03.java
@Test
public void shouldTransferFunds() {
    processor.setAccountBalance("me", 150);
    processor.setAccountBalance("you", 20);
    processor.processTransaction(newTransaction().setSender("me").setRecipient("you").setAmount(100));
    assertThat(processor.getAccountBalance("me")).isEqualTo(50);
    assertThat(processor.getAccountBalance("you")).isEqualTo(120);
}

@Test
public void shouldNotPerformInvalidTransactions() {
    processor.setAccountBalance("me", 50);
    processor.setAccountBalance("you", 20);
    processor.processTransaction(newTransaction().setSender("me").setRecipient("you").setAmount(100));
    assertThat(processor.getAccountBalance("me")).isEqualTo(50);
    assertThat(processor.getAccountBalance("you")).isEqualTo(20);
}

File: assets/code-variants/ch12/03.py
def test_should_transfer_funds():
    processor.set_account_balance("me", 150)
    processor.set_account_balance("you", 20)
    processor.process_transaction(new_transaction(sender="me", recipient="you", amount=100))
    assert processor.get_account_balance("me") == 50
    assert processor.get_account_balance("you") == 120

def test_should_not_perform_invalid_transactions():
    processor.set_account_balance("me", 50)
    processor.set_account_balance("you", 20)
    processor.process_transaction(new_transaction(sender="me", recipient="you", amount=100))
    assert processor.get_account_balance("me") == 50
    assert processor.get_account_balance("you") == 20
